easybuild/easyblocks/l/ls_dyna.py

```diff
@@ -0,0 +1,106 @@
+##
+# Copyright 2009-2021 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBlock for LS-DYNA
+
+@author: James Carpenter (University of Birmingham)
+"""
+import os
+import re
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.run import run_cmd
+
+
+class EB_LS_minus_DYNA(EasyBlock):
+    """Support for installing LS-DYNA.
+    Unpack the sources and execute the install command
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Constructor, adds extra class variables."""
+        super(EB_LS_minus_DYNA, self).__init__(*args, **kwargs)
+        self.build_sources = []
+
+    def extract_step(self):
+        """Unpack the source"""
+        EasyBlock.extract_step(self)
+
+    def configure_step(self):
+        """No configuration, this is binary software"""
+        pass
+
+    def build_step(self):
+        """No compilation, this is binary software"""
+        pass
+
+    def install_step(self):
+        """Install the SMP versions and the best matches for MPP versions"""
+        # Find CPU info
+        cmd = "cat /proc/cpuinfo"
+        (cpuinfo, _) = run_cmd(cmd, log_all=True, simple=False)
+        # List capabilities to check against. N.B. order is important!
+        cpu_caps_regex = [r"avx512", r"avx2", r"sse2"]
+        best_cpu_caps = None
+        for regex in cpu_caps_regex:
+            if re.search(regex, cpuinfo):
+                best_cpu_caps = regex
+                # Exit loop as soon as we match
+                break
+        if not best_cpu_caps:
+            raise EasyBuildError("Unable to determine best vectorisation capabilities for cpu")
```

Comment:
```suggestion

        if not best_cpu_caps:
            raise EasyBuildError("Unable to determine best vectorisation capabilities for cpu")

```

Extra line breaks to make this easier to read.